.github/workflows/demo-windows.yml

```diff
@@ -8,9 +8,38 @@ jobs:
     name: Build
     runs-on: windows-2022
     steps:
-      - name: Switch to msys2
-        shell: powershell
-        run: echo "C:\msys64\usr\bin" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
+      - name: Setup MSVC
+        run: |
+          $script=@'
+          Push-Location "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\Build"
+          & "C:\Windows\System32\cmd.exe" /c "vcvars64.bat & set" |
+          ForEach-Object {
+            if ($_ -match "=") {
+              $v = $_.split("=", 2)
+          
+              if ($($v[0]) -match "Path") {
+                echo "$($v[0])=$($v[1]);C:\msys64\usr\bin" | Out-File -FilePath $Env:GITHUB_ENV -Encoding utf8 -Append
```

Comment:
Cleanup this case. We are adding msys64 at the very last in the `Path` such that binaries like `link.exe` have a higher priority